bioptim/gui/plot.py

```diff
@@ -591,6 +611,23 @@ def update_data(self, v: dict):
                         )
                         raise ValueError(f"Wrong dimensions for plot {key}. Got " f"{val}" f", but expected {y.shape}")
                     self.__append_to_ydata(y)
+                else:
+                    y = np.empty((self.variable_sizes[i][key], len(self.t[i])))
+                    y.fill(np.nan)
+                    try:
+                        y[:, :] = self.plot_func[key][i].function(
+                            state[:, ::step_size], control, data_params_in_dyn, **self.plot_func[key][i].parameters
+                        )
+                    except ValueError:
+                        val = (
+                            self.plot_func[key][i]
+                                .function(
+                                state[:, ::step_size], control, data_params_in_dyn, **self.plot_func[key][i].parameters
+                            )
+                                .shape
+                        )
+                        raise ValueError(f"Wrong dimensions for plot {key}. Got " f"{val}" f", but expected {y.shape}")
+                    self.__append_to_ydata(y)
```

Comment:
I feel this section and the previous one are exact copy except for `len(self.t[i])` vs `len(self.t_to_plot[i])`. If I am not mistaken, please use the else as previous and a if only a that particular line to reduce code duplication